qcommon/src/org/qcmg/common/util/ChrPositionUtils.java

```diff
@@ -30,6 +30,59 @@ public static boolean doChrPositionsOverlap(ChrPosition a, ChrPosition b) {
         return doChrPositionsOverlap(a, b, 0);
     }
 
+    /**
+     * convert ChrPosition to a long. Will only examine the contig and start position
+     *
+     */
+    public static long convertContigAndPositionToLong(String contig, int position) {
+        long l = (long) convertContigNameToInt(contig) << 32;
+        return l + position;
+    }
+
+    public static ChrPosition convertLongToChrPosition(long l) {
+        int contig = (int) (l >> 32);
+        int position = (int) l;
+
+        if (contig == 23) {
+            return ChrPointPosition.valueOf("X", position);
+        } else if (contig == 24) {
+            return ChrPointPosition.valueOf("Y", position);
+        } else if (contig == 25) {
+            return ChrPointPosition.valueOf("M", position);
+        } else if (contig > 25) {
+            return ChrPointPosition.valueOf("-1", position);
+        }
+        return ChrPointPosition.valueOf(Integer.toString(contig), position);
+    }
+
+    /**
+     * Converts a contig name to an integer representing the contig.
+     * Assumes that the contig name is either a number or a string that does not start with "chr".
+     *
+     * @param contigName the name of the contig
+     * @return an integer representation of the contig
+     */
+    public static int convertContigNameToInt(String contigName) {
+        if (null == contigName || contigName.isEmpty()) {
+            throw new IllegalArgumentException("null or empty contig name supplied to convertContigNameToInt");
+        }
+        int i = Character.isDigit(contigName.charAt(0)) ? Integer.parseInt(contigName) : -1;
+        if (i > -1) {
+            return i;
+        }
+
+        if (contigName.length() > 3 && contigName.startsWith("chr")) {
+            return convertContigNameToInt(contigName.substring(3));
+        }
+
+        return switch (contigName) {
+            case "X" -> 23;
+            case "Y" -> 24;
+            case "M" -> 25;
```

Comment:
handle MT (from chrMT) as well?

Reply:
yep. It was left out because I think our version of the GRCh38 ref uses M rather than MT, but can do no harm having it in there and will no doubt save some errors in the future